Makefile

```diff
@@ -117,6 +117,20 @@ ${NORMALIZATION_GRPC}: ${NORMALIZATION_PROTO} ${PROTOC_GEN_GRPC_SWIFT}
 .PHONY:
 generate-normalization: ${NORMALIZATION_PB} ${NORMALIZATION_GRPC}
 
+SERIALIZATION_PATH=Tests/GRPCTests/Codegen/Serialization/
+
+# For serialization we'll set the ReflectionService option to true.
+${SERIALIZATION_GRPC}: ${ECHO_PROTO} ${PROTOC_GEN_GRPC_SWIFT}
+	protoc $< \
+		--proto_path=$(dir $<) \
+		--plugin=${PROTOC_GEN_GRPC_SWIFT} \
+		--grpc-swift_opt=ReflectionService=true \
+		--grpc-swift_out=$(dir ${SERIALIZATION_PATH})
+
+# Generates protobufs and gRPC client and server for the Serialization test
+.PHONY:
+test-serialization: ${SERIALIZATION_GRPC}
```

Comment:
I don't think this will work correctly... 

`${SERIALIZATION_GRPC}` is the "target", which is the name of the object to make, but `SERIALIZATION_GRPC` isn't defined anywhere. It should be defined as "Tests/GRPCTests/Codegen/Serialization/whatever-the-generated-file-is-named".

`${ECHO_PROTO}` and `${PROTOC_GEN_GRPC_SWIFT}` are pre-requisites: i.e. targets which must exist before this target can be created. This part is right: we want echo.proto to exist and the grpc-swift plugin for protoc.

In the options we only specify `ReflectionService=true`, but we also don't want to generate the client/server code using this rule so we should set `Client=false` and `Server=false` too.

We should also think about naming a bit more: all the other rules for generating code start with `generate-`. I think we should name the rule `generate-reflection-data`.

Reply:
This might be useful: https://makefiletutorial.com